tests/test_parts.py

```diff
@@ -672,5 +675,139 @@ def helper_test_get_part_location(parts):
+        )
+
+    try:
+        parts.update_parts_list_properties(
+            "Test",
+            "CCA_name",
+            [
+                {
+                    "ref_des": ["C1"],
+                    "properties": [{"name": "partType", "value": "RESISTOR"}],
+                    "test": "test",
+                }
+            ],
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockUpdatePartsListPropertiesError as e:
+        assert str(e.str_itr()) == (
+            "['Update parts list properties error: Number of elements (3) "
+            "is wrong for part list property 0.']"
+        )
+
+    try:
+        parts.update_parts_list_properties(
+            "Test",
+            "CCA_name",
+            [{"ref_des": "C1", "properties": [{"name": "partType", "value": "RESISTOR"}]}],
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockUpdatePartsListPropertiesError as e:
+        assert str(e.str_itr()) == (
+            "['Update parts list properties error: ref_des is not a list "
+            "for parts list property 0.']"
+        )
+
+    try:
+        parts.update_parts_list_properties(
+            "Test",
+            "CCA_name",
+            [
+                {
+                    "ref_des": ["C1"],
+                    "properties": [{"name": "partType", "value": "RESISTOR", "test": "test"}],
+                }
+            ],
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockUpdatePartsListPropertiesError as e:
+        assert str(e.str_itr()) == (
+            "['Update parts list properties error: Number of elements (3) "
+            "is wrong for property 0.']"
+        )
+
+    try:
+        parts.update_parts_list_properties(
+            "Test",
+            "CCA_name",
+            [{"ref_des": ["C1"], "properties": [{"name": "", "value": "RESISTOR"}]}],
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockUpdatePartsListPropertiesError as e:
+        assert str(e.str_itr()) == (
+            "['Update parts list properties error: Name is required for property 0.']"
+        )
+
+    try:
+        parts.update_parts_list_properties(
+            "Test",
+            "CCA_name",
+            [{"ref_des": ["C1"], "properties": [{"name": "partType", "value": ""}]}],
+        )
+        pytest.fail("No exception raised when using an invalid parameter")
+    except SherlockUpdatePartsListPropertiesError as e:
+        assert str(e.str_itr()) == (
+            "['Update parts list properties error: Value is required for property 0.']"
+        )
+
+    if not parts._is_connection_up():
+        return
+
+    try:
+        parts.update_parts_list_properties(
+            "Test",
+            "CCA_name",
+            [{"ref_des": ["C1"], "properties": [{"name": "partType", "value": "RESISTOR"}]}],
```

Comment:
It is unclear which parameter is intended to be invalid.
Other tests use "Invalid CCA" for the CCA name, and "Tutorial Project" for the project name. It would be better if these were consistent.